core/src/main/java/org/apache/carbondata/core/util/CleanFilesUtil.java

```diff
@@ -0,0 +1,179 @@
+          // copy the complete segment to the trash folder
+          TrashUtil.copySegmentToTrash(segmentLocation, CarbonTablePath.getTrashFolderPath(
+              carbonTable.getTablePath()) + CarbonCommonConstants.FILE_SEPARATOR +
+              timeStampForTrashFolder + CarbonCommonConstants.FILE_SEPARATOR + CarbonTablePath
+              .SEGMENT_PREFIX + segmentNumber);
+          // Deleting the stale Segment folders.
+          try {
+            CarbonUtil.deleteFoldersAndFiles(segmentLocation);
+          } catch (IOException | InterruptedException e) {
+            LOGGER.error("Unable to delete the segment: " + segmentNumber + " from after moving" +
+                " it to the trash folder : " + e.getMessage(), e);
+          }
+          // delete the segment file as well
+          FileFactory.deleteFile(CarbonTablePath.getSegmentFilePath(carbonTable.getTablePath(),
+              staleSegment));
+        }
+      }
+      staleSegments.clear();
+    }
+  }
+
+  /**
+   * This method will clean all the stale segments for partition table, delete the source folders
+   * after copying the data to the trash and also remove the .segment files of the stale segments
+   */
+  public static void cleanStaleSegmentsForPartitionTable(CarbonTable carbonTable)
+    throws IOException {
+    long timeStampForTrashFolder = System.currentTimeMillis();
+    List<String> staleSegments = getStaleSegments(carbonTable);
+    if (staleSegments.size() > 0) {
+      for (String staleSegment : staleSegments) {
+        String segmentNumber = staleSegment.split(CarbonCommonConstants.UNDERSCORE)[0];
+        // for each segment we get the indexfile first, then we get the carbondata file. Move both
+        // of those to trash folder
+        List<CarbonFile> filesToDelete = new ArrayList<>();
+        SegmentFileStore fileStore = new SegmentFileStore(carbonTable.getTablePath(),
+            staleSegment);
+        List<String> indexOrMergeFiles = fileStore.readIndexFiles(SegmentStatus.SUCCESS, true,
+            FileFactory.getConfiguration());
+        for (String file : indexOrMergeFiles) {
+          // copy the index or merge file to the trash folder
+          TrashUtil.copyFileToTrashFolder(file, CarbonTablePath.getTrashFolderPath(carbonTable
+              .getTablePath()) + CarbonCommonConstants.FILE_SEPARATOR + timeStampForTrashFolder +
+              CarbonCommonConstants.FILE_SEPARATOR + CarbonTablePath.SEGMENT_PREFIX +
+              segmentNumber);
+          filesToDelete.add(FileFactory.getCarbonFile(file));
+        }
+        // get carbondata files from here
+        Map<String, List<String>> indexFilesMap = fileStore.getIndexFilesMap();
+        for (Map.Entry<String, List<String>> entry : indexFilesMap.entrySet()) {
+          for (String file : entry.getValue()) {
+            // copy the carbondata file to trash
+            TrashUtil.copyFileToTrashFolder(file, CarbonTablePath.getTrashFolderPath(carbonTable
```

Comment:
here also, need to fail clean files operation if anything fails inside. In trash folder util there is currently method to remove file, a method to remove the segment folder(only valid for normal table flow). I can add it to a list here and use the copyTrashforFile method, but it will be same. Let me know if that part i have to change

Reply:
okay, collected all the file paths to a list and sending it to trash one by one